nemo/lightning/pytorch/callbacks/megatron_comm_overlap.py

```diff
@@ -209,6 +209,32 @@ def _override_user_cfgs(self, comm_overlap_cfg):
 
         return comm_overlap_cfg
 
+    def _set_num_cuda_device_max_connections(self):
+        import os
+
+        import torch
+
+        from nemo.utils import AppState
+
+        app_state = AppState()
+        tp_size = app_state.tensor_model_parallel_size
+        cp_size = app_state.context_parallel_size
+        dp_size = app_state.data_parallel_size
+        pp_size = app_state.pipeline_model_parallel_size
+        major, _ = torch.cuda.get_device_capability()
+        if major > 9:
+            if (tp_size > 1 or cp_size > 1) and (dp_size > 1 or pp_size > 1):
+                # Default is 8, but for this case, we need extra connections
+                # to avoid serialization of streams
```

Comment:
`Default is 8, but for this case, we need extra connections to avoid serialization of streams`
to 
`We need extra connections to avoid serialization of streams, so we use the max connections of 32 instead of the default device connection of 8. `